api/apps/v1alpha1/nemo_evaluator_types.go

```diff
@@ -0,0 +1,764 @@
+	NemoEvaluatorStatusFailed = "Failed"
+)
+
+// NemoEvaluatorSpec defines the desired state of NemoEvaluator
+type NemoEvaluatorSpec struct {
+	Image   Image           `json:"image,omitempty"`
+	Command []string        `json:"command,omitempty"`
+	Args    []string        `json:"args,omitempty"`
+	Env     []corev1.EnvVar `json:"env,omitempty"`
+	// The name of an secret that contains authn for the NGC NIM service API
+	Labels         map[string]string            `json:"labels,omitempty"`
+	Annotations    map[string]string            `json:"annotations,omitempty"`
+	NodeSelector   map[string]string            `json:"nodeSelector,omitempty"`
+	Tolerations    []corev1.Toleration          `json:"tolerations,omitempty"`
+	PodAffinity    *corev1.PodAffinity          `json:"podAffinity,omitempty"`
+	Resources      *corev1.ResourceRequirements `json:"resources,omitempty"`
+	Expose         Expose                       `json:"expose,omitempty"`
+	LivenessProbe  Probe                        `json:"livenessProbe,omitempty"`
+	ReadinessProbe Probe                        `json:"readinessProbe,omitempty"`
+	StartupProbe   Probe                        `json:"startupProbe,omitempty"`
+	Scale          Autoscaling                  `json:"scale,omitempty"`
+	Metrics        Metrics                      `json:"metrics,omitempty"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default:=1
+	Replicas     int    `json:"replicas,omitempty"`
+	UserID       *int64 `json:"userID,omitempty"`
+	GroupID      *int64 `json:"groupID,omitempty"`
+	RuntimeClass string `json:"runtimeClass,omitempty"`
+
+	Mongodb       Mongodb       `json:"mongodb"`
+	ArgoWorkFlows ArgoWorkFlows `json:"argoWorkFlows"`
+	Milvus        Milvus        `json:"milvus"`
+	DataStore     DataStore     `json:"dataStore"`
+}
+
+// NemoEvaluatorStatus defines the observed state of NemoEvaluator
+type NemoEvaluatorStatus struct {
+	Conditions        []metav1.Condition `json:"conditions,omitempty"`
+	AvailableReplicas int32              `json:"availableReplicas,omitempty"`
+	State             string             `json:"state,omitempty"`
+}
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.state`,priority=0
+// +kubebuilder:printcolumn:name="Age",type="date",format="date-time",JSONPath=".metadata.creationTimestamp",priority=0
+
+// NemoEvaluator is the Schema for the NemoEvaluator API
+type NemoEvaluator struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NemoEvaluatorSpec   `json:"spec,omitempty"`
+	Status NemoEvaluatorStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// NemoEvaluatorList contains a list of NemoEvaluator
+type NemoEvaluatorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NemoEvaluator `json:"items"`
+}
+
+type Mongodb struct {
```

Comment:
All these deps can be pulled into common types file if the params are same between all microservices.

Reply:
Fixed it!